Services/CO.CDP.Forms.WebApi/UseCase/UpdateFormSectionAnswersUseCase.cs

```diff
@@ -23,8 +23,52 @@ public async Task<bool> Execute((Guid formId, Guid sectionId, Guid answerSetId,
         var section = await formRepository.GetSectionAsync(formId, sectionId)
             ?? throw new UnknownSectionException($"Unknown section {sectionId} in form {formId}.");
 
-        var sharedConsent = await formRepository.GetSharedConsentDraftWithAnswersAsync(formId, organisationId)
-                            ?? CreateSharedConsent(organisation, section.Form);
+        var sharedConsent = await formRepository.GetUntrackedSharedConsent(formId, organisationId);
+        if (sharedConsent == null)
+        {
+            sharedConsent = CreateSharedConsent(organisation, section.Form);
+
+            await UpdateOrAddStandardAnswers(formRepository, answerSetId, answers, section, sharedConsent);
+            await formRepository.SaveSharedConsentAsync(sharedConsent);
+
+            return true;
+        }
+
+        if (sharedConsent.SubmissionState == Persistence.SubmissionState.Submitted)
+        {
+            CleanSharedConsent(sharedConsent);
+
+            await UpdateOrAddStandardAnswers(formRepository, answerSetId, answers, section, sharedConsent);
+            await formRepository.SaveSharedConsentAsync(sharedConsent);
+        }
+        else
+        {
+            sharedConsent = await formRepository.GetSharedConsentWithAnswersAsync(formId, organisationId);
+            if (sharedConsent != null)
+            {
+                await UpdateOrAddStandardAnswers(formRepository, answerSetId, answers, section, sharedConsent);
+                await formRepository.SaveSharedConsentAsync(sharedConsent);
+            }
+        }
+
+        return true;
+    }
+
+    private static void CleanSharedConsent(Persistence.SharedConsent sharedConsent)
+    {
+        sharedConsent.Id = default;
+        sharedConsent.Guid = Guid.NewGuid();
+        sharedConsent.SubmittedAt = default;
+        sharedConsent.BookingReference = default;
+        sharedConsent.SubmissionState = default;
+    }
+
+    private async Task UpdateOrAddStandardAnswers(IFormRepository formRepository, Guid answerSetId, List<FormAnswer> answers, Persistence.FormSection section, Persistence.SharedConsent sharedConsent)
+    {
+        if (section.Type != Persistence.FormSectionType.Standard)
+        {
+            return;
+        }
```

Comment:
Won't this prevent adding answers to the Declaration of a shared consent in draft?

Reply:
done.